src/handlers/user-identity-handler.ts

```diff
@@ -1,33 +1,97 @@
+import { decodeJwt, JWTPayload } from "jose";
+
 import logger from "../commons/logger";
-import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
-import { HTTP_MESSAGE_INVALID_REQUEST, HttpCodesEnum } from "../types/constants";
-
-export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
-  logger.debug("Received message", { event: event.body });
-  if (!event.body) {
-    return {
-      statusCode: HttpCodesEnum.BAD_REQUEST,
-      body: JSON.stringify({ message: HTTP_MESSAGE_INVALID_REQUEST }),
-    };
-  }
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { HttpCodesEnum } from "../types/constants";
+import { getConfiguration, getServiceApiKey } from "../types/configuration";
+import { EvcsStoredIdentityResponse, StoredIdentityResponse, UserIdentityDataType } from "../types/interfaces";
+
+interface ErrorResponse {
+  error: string;
+  error_description: string;
+}
+
+export const handler = async (event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> => {
+  logger.addContext(context);
+
   if (!event?.headers?.Authorization) {
-    return {
-      statusCode: HttpCodesEnum.UNAUTHORIZED,
-      body: JSON.stringify({ message: "Request Unauthorized" }),
-    };
+    logger.error("Authorisation header was not included in request");
+    return createErrorResponse(HttpCodesEnum.UNAUTHORIZED);
   }
 
+  const configuration = await getConfiguration();
+  const serviceApiKey = await getServiceApiKey();
+
   try {
-    JSON.parse(JSON.stringify(event.body));
-  } catch {
-    return {
-      statusCode: HttpCodesEnum.BAD_REQUEST,
-      body: JSON.stringify({ message: "Event body is not valid JSON" }),
+    try {
+      getJwtBody(event.headers.Authorization.split(" ").at(1) || ""); // Validate bearer token
+    } catch {
+      logger.error("Error whilst decoding Bearer token body");
+      return createErrorResponse(HttpCodesEnum.UNAUTHORIZED);
+    }
+
+    const result = await fetch(`${configuration.evcsApiUrl}/identity`, {
+      method: "GET",
+      headers: {
+        ...event.headers,
```

Comment:
Just a thought, but are there any headers that might be on the request to SIS that would be irrelevant/confusing to send to the EVCS endpoint?

My understanding is that we're copying all headers across and I worry that there may be a future where we accidentally pass something across that we don't want to. Would it be safer just to copy the `Authorizaton` header over?